src/CalcViewModel/GraphingCalculator/EquationViewModel.cpp

```diff
@@ -1,14 +1,316 @@
 #include "pch.h"
 #include "EquationViewModel.h"
+#include "CalcViewModel\Common\LocalizationSettings.h"
+#include "CalcViewModel\GraphingCalculatorEnums.h"
 
+using namespace CalculatorApp::Common;
+using namespace Graphing;
+using namespace Platform;
+using namespace Platform::Collections;
+using namespace std;
 using namespace Windows::UI;
 using namespace Windows::UI::Xaml;
+using namespace Windows::Foundation::Collections;
 
 namespace CalculatorApp::ViewModel
 {
+    GridDisplayItems::GridDisplayItems()
+        : m_Expression{ "" }
+        , m_Direction{ "" }
+    {
+    }
+
+    KeyGraphFeaturesItem::KeyGraphFeaturesItem()
+        : m_Title{ "" }
+        , m_DisplayItems{ ref new Vector<String ^>() }
+        , m_GridItems{ ref new Vector<GridDisplayItems ^>() }
+        , m_IsText{ false }
+    {
+    }
+
     EquationViewModel::EquationViewModel()
-        : m_LineColor{ nullptr }, m_KeyGraphFeaturesVisibility{ ::Visibility::Collapsed }
+        : m_LineColor{ nullptr }
         , m_Expression{ "" }
+        , m_IsAnalysisUpdated{ false }
+        , m_Domain{ "" }
+        , m_Range{ "" }
+        , m_XIntercept{ "" }
+        , m_YIntercept{ "" }
+        , m_Parity{ -1 }
+        , m_PeriodicityDirection{ -1 }
+        , m_PeriodicityExpression{ "" }
+        , m_Minima{ ref new Vector<String ^>() }
+        , m_Maxima{ ref new Vector<String ^>() }
+        , m_InflectionPoints{ ref new Vector<String ^>() }
+        , m_Monotonicity{ ref new Map<String ^, String ^>() }
+        , m_VerticalAsymptotes{ ref new Vector<String ^>() }
+        , m_HorizontalAsymptotes{ ref new Vector<String ^>() }
+        , m_ObliqueAsymptotes{ ref new Vector<String ^>() }
+        , m_TooComplexFeatures{ -1 }
+        , m_AnalysisError{ 0 }
+        , m_AnalysisErrorVisible{ false }
+        , m_KeyGraphFeaturesItems{ ref new Vector<KeyGraphFeaturesItem ^>() }
+        , m_resourceLoader{ Windows::ApplicationModel::Resources::ResourceLoader::GetForCurrentView() }
+    {
+    }
+
+    void EquationViewModel::OnPropertyChanged(String ^ propertyName)
+    {
+        if (propertyName == L"IsAnalysisUpdated" && IsAnalysisUpdated)
```

Comment:
I think it would be cleaner to simplify this if statement to if (propertyName == L"IsAnalysisUpdated") and call a method called OnIsAnalysisUpdatedChanged()